xcodeproject/xcodeproj/target_dependency.go

```diff
@@ -19,13 +19,8 @@ func parseTargetDependency(id string, objects serialized.Object) (TargetDependen
 		return TargetDependency{}, err
 	}
 
-	target, err := parseTarget(targetID, objects)
```

Comment:
Can you add a test case with more dependent targets?
Just want to be sure that it is guaranteed that:
- We parse all dependent targets, even with this part removed.
- If there is a target dependency that is missing from the project, it is handled gracefully. (At least no additional errors are returned compared to now)

Reply:
For the first point, take a look at `target_test.go`, it has tests for parsing the entire Target structure with the dependent target ID reference as well.

Regarding the rest of the parsing and any missing dependency: It's handled now in `Xcodeproj.DependentTargetsOfTarget()`, which is tested in `xcodeproj_test.go` with 2 dependencies. I also added a new test case for a target that has no dependencies. I'm not sure about the case of an invalid dependency, I think this setup would fail anyway sooner or later in `xcodebuild`. Our old implementation produced an error (target.go line 127), now it returns an empty dependency list. To me it seems a better solution, but I can change my mind.

Reply:
I think it does return a partial dependency list in case of an invalid dependency, right?
Not returning an error is a good idea, as Xcode projects can have missing targets sometimes (Xcode has some logic that recreates some targets if not found).